### PR TITLE
fix: Revert #1244 to address regression w/ Button and DropdownItem

### PR DESCRIPTION
**Pull request by rnicholus**
In #1244, the return type of `Button` and `DropdownItem` was changed to `ReactNode`. This is not an acceptable return type for functional components. The correct return type is `JSX.Element`. This change fixes the regression from that PR (introduced in v0.7.3) by reverting that PR and accounting for related changes that followed that PR. Also fixes https://github.com/themesberg/flowbite-react/issues/962.

Perhaps there is a way to fulfill the goals of #1244 and solve this regression, but I wasn't able to find time to address that, and #1244 seems flawed enough that perhaps it just needs to be redesigned.

### CODE REVIEW

**Comment by rluders** [comment]
@coderabbitai review

**Comment by rnicholus** [comment]
Uh, whatever AI you’ve unleashed on this PR is pretty low-quality. All I’ve done here is ‘git revert’ 1 set of changes and partially revert another.

**Comment by SutuSebastian** [comment]
> Uh, whatever AI you’ve unleashed on this PR is pretty low-quality. All I’ve done here is ‘git revert’ 1 set of changes and partially revert another.

Yep, I know that, but seems like the AI does not

**Comment by rluders** [comment]
@rnicholus AI is just an tool, we still reviewing as humans. 😁

**Comment by rnicholus** [comment]
Is there anything else i can do to ensure this change is merged & released?

File: components/homepage/components-section.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ export const ComponentsSection: FC = () => {
           ))}
         </div>
         <div className="mb-4 flex w-full justify-center text-center">
+          {/* @ts-expect-error TODO: fix `as` inference */}
```
Comment (nigellima) [inline components/homepage/components-section.tsx:28]: As you are fixing #962, is it necessary to have these @ts-expect-error comments?

File: src/components/Dropdown/Dropdown.spec.tsx [hunk 2]
```diff
@@ -167,7 +167,7 @@ describe('Components / Dropdown', () => {
   });
 
   describe('Dropdown item render', async () => {
-    it('should override Dropdown.Item base component when using `as` prop', async () => {
+    it('should override Dropdownn.Item base component when using `as` prop', async () => {
```
Comment (nigellima) [inline src/components/Dropdown/Dropdown.spec.tsx:170]: Keep the latest changes here

**Review by nigellima — CHANGES_REQUESTED** [review]

File: components/homepage/components-section.tsx (hunk 1 above)
Comment (rnicholus) [reply components/homepage/components-section.tsx:28]: Yes, this is needed since the fix is reverting #1244.

**Review by nigellima — APPROVED** [review]

File: src/components/Dropdown/Dropdown.spec.tsx (hunk 2 above)
Comment (rnicholus) [reply src/components/Dropdown/Dropdown.spec.tsx]: fixed in https://github.com/themesberg/flowbite-react/pull/1298/commits/3292c56f764797bd21f38aa399e590ff2553de59

**Comment by rluders** [comment]
@rnicholus please, no merge with main... rebase it instead!

**Comment by rnicholus** [comment]
> @rnicholus please, no merge with main... rebase it instead!

this can all be squashed on merge, correct?

**Comment by rluders** [comment]
@rnicholus yes, but it is preferable to rebase with the main instead of merging it. But, anyway...

I'm wondering if this PR is actually fixing the issue, and not just reverting for some broken state. Isn't better if we could address the problem with a definitive fix? What would be the pros and cons of accepting this to fix #962 and reverting the fix for #1002 and #1107?

**Comment by rnicholus** [comment]
> @rnicholus yes, but it is preferable to rebase with the main instead of merging it. But, anyway...
> 
> I'm wondering if this PR is actually fixing the issue, and not just reverting for some broken state. Isn't better if we could address the problem with a definitive fix? What would be the pros and cons of accepting this to fix #962 and reverting the fix for #1002 and #1107?

I'm not sure. My motivation here was to fix the typing issue without having to resort to `ts-ignore` in my project, and i suspect a bunch of other users are in the same boat.

>Isn't better if we could address the problem with a definitive fix?

absolutely, but i'm not sure what that fix is as i don't really have the cycles to investigate

**Comment by SutuSebastian** [comment]
Closed in favor of #1308

**Pull request closed without merging**